type alias (of map) does not support indexing

**V version:** 0.1.28
**OS:** Ubuntu 20.04

**What did you do?**
```v
module main

type Test = map[string]string

fn (t Test) get(key string) string {
    return t[key]
}

fn main() {
    t := Test({'c': 'abc'})
    r := t.get('c')
    println(r)
}
```

**What did you expect to see?**
it should work
 
**What did you see instead?**
```
main.v:7:1: error: type `Test` does not support indexing 
    5 | fn (t Test) get(key string) string {
    6 |     return t[key]
    7 | }
      | ^
    8 | 
    9 | fn main() {
```


If i replace `t[key]` with `t.get(key)` i get a segfault.

```v
fn (t Test) get2(key string) string {
	return t.get(key, '')
}
```
```
/home/daniel/.cache/v/main.tmp.c: In function ‘Test_get2’:
/home/daniel/.cache/v/main.tmp.c:7598:34: error: lvalue required as unary ‘&’ operand
 7598 |  return map_get(t, key, (voidptr)&/*qq*/tos_lit(""));
      |                                  ^
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -4601,7 +4601,7 @@ fn (mut c Checker) check_index_type(typ_sym &table.TypeSymbol, index_type table.
 pub fn (mut c Checker) index_expr(mut node ast.IndexExpr) table.Type {
 	typ := c.expr(node.left)
 	node.left_type = typ
-	typ_sym := c.table.get_type_symbol(typ)
+	typ_sym := c.table.get_final_type_symbol(typ)
 	if typ_sym.kind !in [.array, .array_fixed, .string, .map] && !typ.is_ptr() && !(!typ_sym.name[0].is_capital() &&
 		typ_sym.name.ends_with('ptr')) && !typ.has_flag(.variadic) { // byteptr, charptr etc
 		c.error('type `$typ_sym.name` does not support indexing', node.pos)

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -3840,7 +3840,7 @@ fn (mut g Gen) index_expr(node ast.IndexExpr) {
 			g.write(')')
 		}
 		else {
-			sym := g.table.get_type_symbol(node.left_type)
+			sym := g.table.get_final_type_symbol(node.left_type)
 			left_is_ptr := node.left_type.is_ptr()
 			if sym.kind == .array {
 				info := sym.info as table.Array

--- a/vlib/v/table/table.v
+++ b/vlib/v/table/table.v
@@ -638,7 +638,7 @@ pub fn (mut t Table) add_placeholder_type(name string, language Language) int {
 
 [inline]
 pub fn (t &Table) value_type(typ Type) Type {
-	typ_sym := t.get_type_symbol(typ)
+	typ_sym := t.get_final_type_symbol(typ)
 	if typ.has_flag(.variadic) {
 		// ...string => string
 		// return typ.clear_flag(.variadic)

--- a/vlib/v/tests/array_type_alias_test.v
+++ b/vlib/v/tests/array_type_alias_test.v
@@ -1,0 +1,6 @@
+type Test = []int
+
+fn test_index() {
+    t := Test([2,4])
+    assert t[1] == 4
+}

--- a/vlib/v/tests/map_type_alias_test.v
+++ b/vlib/v/tests/map_type_alias_test.v
@@ -1,11 +1,6 @@
 type Test = map[string]string
 
-fn (t Test) get(key string) string {
-    return t[key]
-}
-
-fn main() {
+fn test_index() {
     t := Test({'c': 'abc'})
-    r := t.get('c')
-    println(r)
+    assert t['c'] == 'abc'
 }

--- a/vlib/v/tests/map_type_alias_test.v
+++ b/vlib/v/tests/map_type_alias_test.v
@@ -1,0 +1,11 @@
+type Test = map[string]string
+
+fn (t Test) get(key string) string {
+    return t[key]
+}
+
+fn main() {
+    t := Test({'c': 'abc'})
+    r := t.get('c')
+    println(r)
+}

